py/objint_mpz.c

```diff
@@ -411,6 +411,24 @@ mp_int_t mp_obj_int_get_checked(mp_const_obj_t self_in) {
     }
 }
 
+mp_uint_t mp_obj_int_get_uint_checked(mp_const_obj_t self_in) {
+    if (mp_obj_is_small_int(self_in)) {
+        if (MP_OBJ_SMALL_INT_VALUE(self_in) < 0) {
+            goto overflow;
+        }
+        return MP_OBJ_SMALL_INT_VALUE(self_in);
+    } else {
+        const mp_obj_int_t *self = MP_OBJ_TO_PTR(self_in);
+        mp_uint_t value;
+        if (mpz_as_uint_checked(&self->mpz, &value)) {
+            return value;
+        } else {
+overflow:
```

Comment:
To avoid labels and goto (which is nice if possible) this function could be easily restructured so that it falls through to the bottom if there's an overflow, and the `mp_raise_msg` is at the bottom.  Please try to do that.

Reply:
You are absolutely right. I dunno why I've structured it with a `goto` in the first place. I think it's a remnant of having a single function with `mp_obj_int_get_checked`.